validate/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/internal/version"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	flagVersion bool
+	rootCmd     = &cobra.Command{
+		Use:   "ignvalidate [CONFIG.IGN]",
+		Short: "ignvalidate will validate Ignition configs",
+		Run:   runIgnValidate,
+	}
+)
+
+func main() {
+	rootCmd.Flags().BoolVar(&flagVersion, "version", false, "print the version of ignvalidate")
+	rootCmd.Execute()
+}
+
+func stdout(format string, a ...interface{}) {
+	fmt.Fprintf(os.Stdout, strings.TrimSpace(format)+"\n", a...)
+}
+
+func stderr(format string, a ...interface{}) {
+	fmt.Fprintf(os.Stderr, strings.TrimSpace(format)+"\n", a...)
+}
+
+func die(format string, a ...interface{}) {
+	stderr(format, a...)
+	os.Exit(1)
+}
+
+func runIgnValidate(cmd *cobra.Command, args []string) {
+	if flagVersion {
+		stdout(version.String)
+		return
+	}
+	if len(args) != 1 {
+		cmd.Usage()
+		os.Exit(1)
+	}
+	var in io.Reader
+	if args[0] == "-" {
```

Comment:
nit: I would declare the `[]byte` and have each if block fill it with `ioutil.RealAll` or `ioutil.ReadFile`. It needs a little less error handling.